### PR TITLE
Fix memleak in tcp server by wiping priority queues on deletion

### PR DESCRIPTION
**Pull request by zugz**
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1216)

### CODE REVIEW

File: toxcore/TCP_server.c [hunk 1]
```diff
@@ -222,6 +222,23 @@ static int add_accepted(TCP_Server *tcp_server, const Mono_Time *mono_time, cons
     return index;
 }
 
+static void wipe_priority_list(TCP_Secure_Connection *con)
```
Comment (iphydf) [inline toxcore/TCP_server.c]: It'll need to be called something else. https://travis-ci.org/TokTok/c-toxcore/jobs/434939506#L1504

**Comment by kpp** [comment]
https://github.com/tox-rs/tox is your best friend =) The second 0day this year.

**Comment by zugz** [comment]
This doesn't actually fix all the potential memleaks yet. e.g. in kill_TCP_server the TCP_Secure_Connections aren't wiped, so there could be unfreed priority queues. Similarly in realloc_connection. I'm also worried about the potential for double frees. This is going to take some care... is there anyone who's actually familiar with TCP_server.c and wouldn't mind helping out?

**Comment by nurupo** [comment]
@zugz what do you mean by "non-verbally"? How exactly did he communicate? Was it by taking down 3/4 of bootstrap nodes? Because if so, then that's a rather malicious way of communicating and tox-rs ain't no friend.

**Comment by zugz** [comment]
* Saturday, 2018-09-29 at 12:36 -0700 - nurupo <notifications@github.com>:

>@zugz what do you mean by "non-verbally"? How exactly did he 
>communicate? Was it by taking down 3/4 of bootstrap nodes?

That is what I meant.

**Comment by kpp** [comment]
@nurupo I sent you a request to add our node written in Rust without this vulnerability to http://nodes.tox.chat but you did not add it. Does it mean there is a censorship in OpenSource development of tox?

**Comment by irungentoo** [comment]
This is how I fixed it: https://github.com/irungentoo/toxcore/commit/bf69b54f64003d160d759068f4816b2d9b2e1e21

**Pull request merged**